backend/migration_service.py

```
@@ -26,6 +27,18 @@ async def start_migration(self, request: MigrationRequest) -> str:
         
+        # Validate all input parameters for security
+        try:
+            SecurityUtils.validate_migration_request(
+                request.compose_dataset,
+                request.target_host,
+                request.ssh_user,
+                request.ssh_port,
+                request.target_base_path
+            )
+        except SecurityValidationError as e:
+            raise Exception(f"Security validation failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)